### PR TITLE
[input_common] Fixed thread hang

### PR DESCRIPTION
**Pull request by VPeruS**
Partially reverted b01aa72, thread was unable to change status to "Completed".  Calibration is supposed to have multiple request-response exchanges, because of that we cannot store calibration's temporary data inside of callback's local variables.
Added small test for verification.

### CODE REVIEW

**Comment by FernandoS27** [comment]
@german77 could you review this?

**Comment by vonchenplus** [comment]
This looks like only a partial rollback of PR #7207, have you tested the compilation of VS 2022?

**Comment by VPeruS** [comment]
No, I haven't tested compilation with VS2022

**Comment by german77** [comment]
This fixes issues with mapping touch. But we need to verify that compiles on vs2022. I will approve once I manage to do that.

**Comment by german77** [comment]
Could you rebase. Files changed a bit

**Review by german77 — APPROVED** [review]

File: src/input_common/drivers/udp_client.cpp [hunk 1]
```diff
@@ -536,42 +536,46 @@ CalibrationConfigurationJob::CalibrationConfigurationJob(
     std::function<void(u16, u16, u16, u16)> data_callback) {
 
     std::thread([=, this] {
+        constexpr u16 CALIBRATION_THRESHOLD = 100;
```
Comment (ameerj) [inline src/input_common/drivers/udp_client.cpp:539]: This needs to be declared in the lambda body to avoid a VS2022 compilation error.

Comment (VPeruS) [reply src/input_common/drivers/udp_client.cpp]: Done

**Review by ameerj — APPROVED** [review]

**Review by bunnei — APPROVED** [review]

**Comment by bunnei** [comment]
LGTM, thank you!

**Pull request merged**